Integrations/integration-DemistoLocking.yml

```diff
@@ -0,0 +1,153 @@
+description: Locking mechanism that prevents concurrent execution of different tasks
+detaileddescription: "The Demisto Locking integration allows prevention of concurrent
+  execution of scripts or commands, using a wait-lock-release flow (mutex). \nUse
+  the lock name argument to support multiple locks in different flows."
+configuration:
+- display: Default timeout for wait on locks to be released
+  name: timeout
+  defaultvalue: "60"
+  type: 0
+  required: true
+script:
+  script: |
+    function guid() {
+      function s4() {
+        return Math.floor((1 + Math.random()) * 0x10000)
+          .toString(16)
+          .substring(1);
+      }
+      return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
+    }
+
+    var setLock = function(guid, info) {
+        var integrationContext = getIntegrationContext();
+        integrationContext[lockName] = {guid: guid, info: info};
+        setIntegrationContext(integrationContext);
+    };
+    var getLock = function() {
+        var integrationContext = getIntegrationContext();
+        if (!integrationContext[lockName]) {
+            integrationContext[lockName] = {};
+        }
+        return integrationContext[lockName];
+    };
+
+    var lockName = (args.name) ? args.name : 'Default';
+
+    switch (command) {
+        case 'test-module':
+            return 'ok';
+
+        case 'demisto-lock-get':
+            var lockTimeout = (args.timeout) ? args.timeout : params.timeout;
+            var lockInfo = 'Locked by incident #' + incidents[0].id + '.';
+            lockInfo += (args.info) ? ' Additional info: ' + args.info :'';
+
+            var guid = guid();
+            var time = 0;
+            var lock = getLock();
+
+            while (lock.guid !== guid && time++ < lockTimeout) {
+                wait(1);
+                lock = getLock();
+                if (lock.guid === guid) {
+                    continue;
+                }
+                if (!lock.guid) {
+                    setLock(guid, lockInfo);
+                }
+            }
+
+            if (lock.guid === guid) {
+                var md = '### Demisto Locking Mechanism\n';
+                md += 'Lock acquired successfully\n';
+                md += 'GUID: ' + guid;
+                return { ContentsFormat: formats.markdown, Type: entryTypes.note, Contents: md } ;
```

Comment:
do you want to add outputs? so users will be able to use it after in their playbook.

Reply:
Let's meet half-way. I'll add the returned object. this way, i'm not messing up the context, but it is easy to extend-context.